### PR TITLE
Allow admin to hide and move navigation menu items

### PR DESCRIPTION
**Pull request by foxerest**
### Summary of changes:

- Add fields isHidden and orderIndex to Category, Subcategory and Team entities
- Add edit methods to CategoriesController and SubcategoriesController
- Update edit method of TeamsController

[User Story Requirements SPHUB-55](https://github.com/dark-side/lanthanum/tree/master/sports_hub_portal/web_application_features/maintain_navigation/user_stories/hide_show_navigation_items)
[User Story Requirements SPHUB-56](https://github.com/dark-side/lanthanum/tree/master/sports_hub_portal/web_application_features/maintain_navigation/user_stories/move_and_order_navigation_items)

### CODE REVIEW

File: src/SportsHub.Business.Tests/Services/SubcategoryServiceTests.cs [hunk 1]
```diff
@@ -42,7 +43,7 @@ public async Task GetSubcategoryByIdAsync_WhenIdIsValid_ReturnsExpectedSubcatego
             // Arrange
             var expectedSubcategoryId = Guid.NewGuid();
 
-            var expectedSubcategory = new Subcategory(name: "Name", Guid.NewGuid());
+            var expectedSubcategory = new Subcategory {Name = "Name", CategoryId = Guid.NewGuid()};
```
Comment (NotJediYet) [inline src/SportsHub.Business.Tests/Services/SubcategoryServiceTests.cs]: Add indents before and after curly braces, please
```suggestion
            var expectedSubcategory = new Subcategory { Name = "Name", CategoryId = Guid.NewGuid() };
```

File: src/SportsHub.Business/Services/Abstraction/ISubcategoryService.cs [hunk 2]
```diff
@@ -9,11 +10,15 @@ public interface ISubcategoryService
...
         
         Task CreateSubcategoryAsync(string subcategoryName, Guid categoryId);
-        
+
+        Task<Guid> GetSubcategoryIdByNameAsync(string subcategoryName);
+
         Task<bool> DoesSubcategoryAlreadyExistByNameAsync(string subcategoryName);
 
-        Task<bool> DoesSubcategoryAlreadyExistByIdAsync(Guid id);
+        Task<bool> DoesSubcategoryAlreadyExistByIdAsync(Guid id);
```
Comment (NotJediYet) [inline src/SportsHub.Business/Services/Abstraction/ISubcategoryService.cs]:
```suggestion
        Task<bool> DoesSubcategoryAlreadyExistByIdAsync(Guid id);
```

File: src/SportsHub.Infrastructure/Repositories/CategoryRepository.cs [hunk 3]
```diff
@@ -31,6 +31,30 @@ public async Task AddCategoryAsync(Category category)
...
+        public async Task<Guid> GetCategoryIdByNameAsync(string categoryName)
+        {
+            var categories = await _context.Categories.ToListAsync();
+            var foundCategory = categories.Find(category => category.Name == categoryName);
+
+            if (foundCategory == null)
+            {
+                return Guid.Empty;
+            }
+            else return foundCategory.Id;
```
Comment (NotJediYet) [inline src/SportsHub.Infrastructure/Repositories/CategoryRepository.cs]:
```suggestion
            var category = await _context.Categories.Find(category => category.Name == categoryName);

            if (category == null)
            {
                return Guid.Empty;
            }
            else
            {
                return foundCategory.Id;
            }
```

Also, consider creating simply `GetByName()` instead

File: src/SportsHub.Web/Validators/EditCategoryModelValidator.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        private async Task<bool> DoesCategoryNameIsUniqueAsync(string categoryName, Guid Id)
+        {
+            var result = await _categoryService.GetCategoryIdByNameAsync(categoryName);
+
+            if (result == Guid.Empty)
+            {
+                return true;
+            }
+            else if (result == Id) { return true; }
+            else return false;
```
Comment (NotJediYet) [inline src/SportsHub.Web/Validators/EditCategoryModelValidator.cs:38]:
```suggestion
            else if (result == Id) 
            { 
                return true;
            }
            else
            {
                return false;
            }
```

File: src/SportsHub.Web/Validators/EditSubcategoryModelValidator.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        private async Task<bool> DoesSubcategoryNameIsUniqueAsync(string subcategoryName, Guid Id)
+        {
+            var result = await _subcategoryService.GetSubcategoryIdByNameAsync(subcategoryName);
+
+            if (result == Guid.Empty)
+            {
+                return true;
+            }
+            else if (result == Id) { return true; }
+            else return false;
```
Comment (NotJediYet) [inline src/SportsHub.Web/Validators/EditSubcategoryModelValidator.cs:47]: same here

File: src/SportsHub.Web/Validators/EditTeamModelValidator.cs [hunk 6]
```diff
@@ -49,7 +48,6 @@ private async Task<bool> DoesTeamNameIsUniqueAsync(string teamName, Guid Id)
             }
             else if (result == Id) { return true; }
             else return false;
```
Comment (NotJediYet) [inline src/SportsHub.Web/Validators/EditTeamModelValidator.cs:50]: same as well

File: src/SportsHub.Infrastructure/Repositories/CategoryRepository.cs (hunk 3 above)
Comment (foxerest) [reply src/SportsHub.Infrastructure/Repositories/CategoryRepository.cs]: great idea, thx

**Review by NotJediYet — APPROVED** [review]

**Review by VitaliiPelekh — APPROVED** [review]

**Review by Yurii-Hubych — APPROVED** [review]

**Pull request merged**